File: Go/Maximum-Subarray.go
package main



func maxSubArray(nums []int) int {
	currentSum, maxSum := 0, nums[0]

	for _, num := range nums {
		if currentSum < 0 {
			currentSum = 0
		}
		currentSum += num
		maxSum = max(currentSum, maxSum)
	}

	return maxSum
}

File: Rust/Maximum-Subarray.rs
pub struct Solution {}

impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut current_sum = 0;
        let mut max_sum = nums[0];

        for num in nums {
            if current_sum < 0 {
                current_sum = 0
            }
            current_sum += num;
            max_sum = i32::max(max_sum, current_sum)
        }

        max_sum
    }
}
fn main() {}
